src/core/src/core_logic/MaintenanceWindow.py

```diff
@@ -59,9 +59,13 @@ def get_remaining_time_in_minutes(self, current_time=None, log_to_stdout=False):
 
         return remaining_time_in_minutes
 
-    def is_package_install_time_available(self, remaining_time_in_minutes=None):
+    def is_packages_install_time_available(self, remaining_time_in_minutes=None, number_of_packages = 1, reboot_manager=None):
         """Check if time still available for package installation"""
-        cutoff_time_in_minutes = Constants.REBOOT_BUFFER_IN_MINUTES + Constants.PACKAGE_INSTALL_EXPECTED_MAX_TIME_IN_MINUTES
+        cutoff_time_in_minutes = Constants.PACKAGE_INSTALL_EXPECTED_MAX_TIME_IN_MINUTES * number_of_packages
+
+        if reboot_manager.reboot_setting != Constants.REBOOT_NEVER:
```

Comment:
Good change. Please start a thread to evaluate and propagate this to the Windows Patch Extension.

Reply:
Started a thread.